intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -86,30 +88,101 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FUNCTION ASSUMES ALL GPUS ARE IDENTICAL WITHIN A NODE\n",
+    "def memory_info():\n",
+    "    pynvml.nvmlInit()\n",
+    "    handle = pynvml.nvmlDeviceGetHandleByIndex(0)\n",
+    "    gpu_mem = pynvml.nvmlDeviceGetMemoryInfo(handle).total\n",
+    "    pynvml.nvmlShutdown()\n",
+    "    return (gpu_mem)\n",
+    "\n",
+    "res = client.run(memory_info)\n",
+    "total_mem = sum(res.values())\n",
+    "min_mem = min(res.values())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ETL memory usage depends on GPU memory per card (not number of GPU's). \n",
+    "# Assumption is 1gb split variant of the data is used\n",
+    "\n",
+    "start_year = None # start year for etl proessing\n",
+    "end_year = None # end year for etl processing (inclusive)\n",
+    "part_count = None # no. of data files to train against\n",
+    "if total_mem >= (512<<30):\n",
+    "    start_year = 2000\n",
+    "    end_year = 2004\n",
+    "    part_count = 16\n",
+    "elif total_mem >= (96<<30):\n",
+    "    start_year = 2000\n",
+    "    end_year = 2003\n",
+    "    part_count = 8\n",
+    "elif total_mem >= (48<<30):\n",
+    "    start_year = 2000\n",
+    "    end_year = 2002\n",
+    "    part_count = 4\n",
+    "elif total_mem >= (16<<30):\n",
+    "    start_year = 2000\n",
+    "    end_year = 2000\n",
+    "    part_count = 2\n",
+    "else:\n",
+    "    start_year = 2000\n",
+    "    end_year = 2000\n",
+    "    part_count = 1"
+   ]
+  },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "#### Define the paths to data and set the size of the dataset"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The cell below serves as a quick start on mortgage data for the year 2000. Visit the [RAPIDS Datasets Homepage](https://docs.rapids.ai/datasets/mortgage-data) and update `data_url` below if you want to try other years"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "# to download data for this notebook, visit https://docs.rapids.ai/datasets/mortgage-data and update the following paths accordingly\n",
-    "acq_data_path = \"/path/to/mortgage/acq\"\n",
-    "perf_data_path = \"/path/to/mortgage/perf\"\n",
-    "col_names_path = \"/path/to/mortgage/names.csv\"\n",
-    "start_year = 2000\n",
-    "end_year = 2016 # end_year is inclusive\n",
-    "part_count = 16 # the number of data files to train against"
+    "data_dir = \"/path/to/mortgage/\"\n",
+    "\n",
+    "if not os.path.exists(data_dir+\"perf\") or not os.path.exists(data_dir+\"acq\"):\n",
+    "    data_url = \"http://rapidsai-data.s3-website.us-east-2.amazonaws.com/notebook-mortgage-data/mortgage_2000_1gb.tgz\"\n",
+    "    !wget -O {data_dir}mortgage.tgz {data_url}\n",
```

Comment:
Replace wget with url lib and tar file to make it more python script friendly (non Jupyter)?

Reply:
Yes, please